Add docs for process queue

DavidFair opened the pull request:
Adds docs for the process queue. This includes:

- Activity Diagrams
- Some rework notes for things we can improve whilst rewriting 

*To test:*
- Open .svg files in a browser to check they make sense / are viewable

DavidFair committed:
Add powerpoint with background and ideas for q. processor

Adds a powerpoint detailing why group adding is difficult and an initial
idea for the queue processor

DavidFair committed:
Add activity diagrams for Queue Processor

Adds activity diagrams for the existing Queue processor flow, to ensure
we still do all the same major steps after refactoring

DavidFair committed:
Add notes for rework on potential improvements

Adds some notes to a md file for various things we can improve whilst
reworking the queue processor

Anders-Markvardsen commented:
Can I suggest change first page of powerpoint Why_cron.odp to include e.g. "Presented in AAI weekly meeting, 19th June 2020, discuss cron vs domain and design"

Anders-Markvardsen commented:
Tested and all .svg files open in browser and display well

Anders-Markvardsen commented:
Can I suggest you an overview sentence or two for each of the diagrams in folder activity_diagrams, that is: which part of the QueueProcessor it mainly refer to and what it is trying to highlight or may show.

For example do this by 1) add a README.md to activity_diagrams and/or 2) add comment within each diagram.

Anders-Markvardsen commented:
I like the rework_notes.md and which also shows not all is carved in stone

DavidFair committed:
Clarify some doc aspects following PR feedback

Clarifies the acitivty diagrams and how they fit into queue processor.

Updates the presentation to note that this was presented during a weekly
meeting for context
